ios/RNSScreenStack.m

```diff
@@ -584,7 +590,10 @@ - (BOOL)gestureRecognizerShouldBegin:(UIGestureRecognizer *)gestureRecognizer
   if (topScreen.fullScreenSwipeEnabled) {
     // we want only `RNSPanGestureRecognizer` to be able to recognize when
     // `fullScreenSwipeEnabled` is set
-    if ([gestureRecognizer isKindOfClass:[RNSPanGestureRecognizer class]]) {
+    if (([gestureRecognizer isKindOfClass:[RNSPanGestureRecognizerHorizontal class]] &&
+         topScreen.swipeDirection == RNSScreenSwipeDirectionHorizontal) ||
+        ([gestureRecognizer isKindOfClass:[RNSPanGestureRecognizerVertical class]] &&
```

Comment:
As vertical swipe doesn't really make sense without `fullScreenSwipeEnabled`:

https://user-images.githubusercontent.com/39658211/151131727-a0047c7d-f10d-4dd8-9e87-be76cb6a4261.mov

we should set it to `true` by default when developer sets `swipeDirection` to `vertical`